python-service/src/pipeline/event_processor.py

```
@@ -0,0 +1,146 @@
+
+
+def dispatch_event(event: BackupEvent, webhook_url: str, secret: str) -> bool:
+    """Send a signed notification to the registered webhook endpoint."""
+    import urllib.request
+    body = format_notification(event).encode()
+    sig  = _sign_payload(body, secret)
+    req  = urllib.request.Request(
+        webhook_url,
+        data=body,
+        headers={"X-Backup-Signature": sig, "Content-Type": "text/plain"},
+    )
```

Ruff (0.15.17)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)